test/utils/fakeIndexedDB.ts

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { IDBDatabase, IDBFactory, IDBRequest } from "fake-indexeddb";
+
+declare global {
+  interface Window {
+    indexedDB: IDBFactory;
+  }
+}
+
+window.indexedDB = new IDBFactory();
+
+const matches = (
+  source: Record<string, any>,
+  obj: Record<string, any>
+): boolean =>
+  Object.keys(source).every(
+    (key) =>
+      Object.prototype.hasOwnProperty.call(obj, key) && obj[key] === source[key]
+  );
+
+const namesDotToUnderscoreMap = new Map([
+  ["iam.id", "iam_id"],
```

Comment:
Note: _strings with dots not permitted. Took me a while to figure out why fake Indexed DB didn't work for "iam.id"
https://stackoverflow.com/questions/74193252/problem-with-saving-record-keys-with-dots-to-indexeddb-dexie-js

Reply:
done